atomic_reactor/plugins/pre_flatpak_create_dockerfile.py

```diff
@@ -0,0 +1,161 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+
+
+Takes a flatpak.json from the git repository, and the module information
+looked up by pre_resolve_module_compose, and outputs
+a Dockerfile that will build a filesystem image for the module
+at /var/tmp/flatpak-build.
+
+Example configuration:
+{
+    'name': 'flatpak_create_dockerfile',
+    'args': {'base_image': 'registry.fedoraproject.org/fedora:latest'}
+}
+"""
+
+import json
+import os
+
+from atomic_reactor.constants import FLATPAK_FILENAME, DOCKERFILE_FILENAME, YUM_REPOS_DIR
+from atomic_reactor.plugin import PreBuildPlugin
+from atomic_reactor.plugins.pre_resolve_module_compose import get_compose_info
+from atomic_reactor.plugins.build_orchestrate_build import override_build_kwarg
+from atomic_reactor.util import render_yum_repo
+
+DOCKERFILE_TEMPLATE = '''FROM {base_image}
+
+LABEL name="{name}"
+LABEL com.redhat.component="{name}"
+LABEL version="{stream}"
+LABEL release="{version}"
+
+RUN dnf -y --nogpgcheck --disablerepo=* --enablerepo=atomic-reactor-module-* \\
+    --installroot=/var/tmp/flatpak-build install {packages}
+COPY cleanup.sh /var/tmp/flatpak-build/tmp/
+RUN chroot /var/tmp/flatpak-build/ /bin/sh /tmp/cleanup.sh
+'''
+
+
+class FlatpakSourceInfo(object):
+    def __init__(self, flatpak_json, compose):
+        self.flatpak_json = flatpak_json
+        self.compose = compose
+
+        mmd = compose.base_module.mmd
+        self.runtime = 'runtime' in mmd.profiles
+
+
+WORKSPACE_SOURCE_KEY = 'source_info'
+
+
+def get_flatpak_source_info(workflow):
+    key = FlatpakCreateDockerfilePlugin.key
+    if key not in workflow.plugin_workspace:
+        return None
+    return workflow.plugin_workspace[key].get(WORKSPACE_SOURCE_KEY, None)
+
+
+def set_flatpak_source_info(workflow, source):
+    key = FlatpakCreateDockerfilePlugin.key
+
+    workflow.plugin_workspace.setdefault(key, {})
+    workspace = workflow.plugin_workspace[key]
+    workspace[WORKSPACE_SOURCE_KEY] = source
+
+
+class FlatpakCreateDockerfilePlugin(PreBuildPlugin):
+    key = "flatpak_create_dockerfile"
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow,
+                 base_image=None):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param base_image: host image used to install packages when creating the Flatpak
+        :
+        """
+        # call parent constructor
+        super(FlatpakCreateDockerfilePlugin, self).__init__(tasker, workflow)
+
+        self.base_image = base_image
+
+    def _load_source(self):
+        flatpak_path = os.path.join(self.workflow.builder.df_dir, FLATPAK_FILENAME)
+        if not os.path.isfile(flatpak_path):
+            raise RuntimeError("{} not found".format(flatpak_path))
```

Comment:
the thought process was that this is the place where you'll get a failure if you try to build a non-flatpak repository with a Dockerfile as a flatpak, so didn't want to let a raw error happen, but agreed that the RuntimeError that I added here is no friendlier than the default IOError - I'll just remove it.